### ISSUE TITLE
Svc::ComQueue fails to compile

### ISSUE THREAD

**Issue by kevin-f-ortega** [nasa/fprime#1791]
| | |
|:---|:---|
|**_F´ Version_**|547197194acff1f99d86e6e441fa9b5b5bd2d60b |
|**_Affected Component_**|  `Svc::ComQueue`  |
---
## Problem Description

When compiling for bare metal, I get this error

```
~/myproject/fprime/Svc/ComQueue/ComQueue.cpp: In member function 'void Svc::ComQueue::configure(Svc::ComQueue::QueueConfigurationTable, NATIVE_UINT_TYPE, Fw::MemAllocator&)':
~/myproject/fprime/Svc/ComQueue/ComQueue.cpp:94:76: error: cannot bind non-const lvalue reference of type 'NATIVE_UINT_TYPE&' {aka 'short unsigned int&'} to an rvalue of type 'NATIVE_UINT_TYPE' {aka 'short unsigned int'}
   94 |     this->m_allocation = this->m_allocator->allocate(this->m_allocationId, totalAllocation, recoverable);
      |                                                                            ^~~~~~~~~~~~~~~
In file included from ~/myproject/fprime/Svc/ComQueue/ComQueue.hpp:14,
                 from ~/myproject/fprime/Svc/ComQueue/ComQueue.cpp:8:
~/myproject/fprime/Fw/Types/MemAllocator.hpp:57:39: note:   initializing argument 2 of 'virtual void* Fw::MemAllocator::allocate(NATIVE_UINT_TYPE, NATIVE_UINT_TYPE&, bool&)'
   57 |                     NATIVE_UINT_TYPE &size,
      |                     ~~~~~~~~~~~~~~~~~~^~~~
make[2]: *** [F-Prime/Svc/ComQueue/CMakeFiles/Svc_ComQueue.dir/build.make:132: F-Prime/Svc/ComQueue/CMakeFiles/Svc_ComQueue.dir/ComQueue.cpp.obj] Error 1
```

## How to Reproduce

1. Compile with a bare metal compiler
2.
3.

## Expected Behavior

I can compile for bare metal.

**Comment by bocchino** [comment]
This should be fixed in #1803.

**Comment by kevin-f-ortega** [comment]
I agree. Thank you @bocchino !
